File: projects/m3/004-string-edit-distance/solutions/jimmyhu/solution-string-edit-distance.py
def edit_distance(s,t):
    # base
    if len(s) == 0:
        return len(t)
    elif len(t) == 0:
        return len(s)
    else:
        cost = 0
    # cost will be 1 only if the last are different
    if s[-1] != t[-1]:
        cost = 1
    d1 = cost + edit_distance(s[:-1],t)
    d2 = cost + edit_distance(s,t[:-1])
    d3 = cost + edit_distance(s[:-1],t[:-1])
    return min([d1,d2,d3])
def main():
    word01 = input('Enter the first word: ').lower()
    word02 = input('Enter the second word: ').lower()
    print(f"the edit distance is: {edit_distance(word01,word02)}")
if __name__ == '__main__':
    main()

File: projects/m3/004-string-edit-distance/solutions/marcolepiane/solution-string-edit-distance.js
let firstString= prompt('insert first string');
let secondString = prompt('insert second string');

function editDistance(firstString, secondString){
    if(firstString.length === 0){
        return secondString.length;
    } else if(secondString.length === 0){
        return firstString.length;
    }else{
        let cost =0;



        if (!firstString.length-1 === secondString.length-1){
            cost = 1;
        }else{
            let d1 =editDistance(firstString.slice(0, -1), secondString +1);
            let d2 =editDistance(firstString, secondString.slice(0, -1) +1);
            let d3 =editDistance(firstString.slice(0,-1), secondString(0, -1), );

            return cost = reduce(d1, d2, d3);
        }
    }
}